File: backend/models/thread.py
from __future__ import annotations

from pydantic import BaseModel


class ThreadSnapshot(BaseModel):
    entity_type: str = "thread"

    tid: int
    pid: int
    name: str
    state: str = "unknown"

    cpu_time_user: float
    cpu_time_system: float
    cpu_percent: float = 0.0

    priority: int = 0
    nice: int = 0
    core_id: int | None = None

    voluntary_ctx_switches: int = 0
    involuntary_ctx_switches: int = 0
    stack_size_bytes: int = 0

File: frontend/src/types/thread.ts
export interface ThreadInfo {
  entity_type: 'thread';
  tid: number;
  pid: number;
  name: string;
  state: string;
  cpu_time_user: number;
  cpu_time_system: number;
  cpu_percent: number;
  priority: number;
  nice: number;
  core_id: number | null;
  voluntary_ctx_switches: number;
  involuntary_ctx_switches: number;
  stack_size_bytes: number;
}
